app/main.py

```diff
@@ -1 +1,17 @@
-# write your code here
+from app.errors import VaccineError, NotWearingMaskError
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
Returning immediately upon encountering a `VaccineError` means that only the first friend's vaccination status is checked. If the task requires checking all friends, consider accumulating results or errors for all friends before returning a message.